### ISSUE TITLE
Component.uses() generates null type safety warning

### ISSUE THREAD

**Issue by KKorzuno** [structurizr/java#188]
Component.uses() always generates a warning stating that "Null type safety: The expression of type 'Component' needs unchecked conversion to conform to '@Nonnull Component". No other layer (e.g. Container.uses()) generates that message.

**Comment by simonbrowndotje** [comment]
Is this a runtime error, or something you're seeing via your IDE. If the latter, which IDE are you using?

**Comment by KKorzuno** [comment]
Yes, thanks. Needed to learn about nullchecks etc. Last time I used java @Nonnull didn't even exist. 
I am generating components dynamically and the rest statically, thus no "null" issues there.

**Comment by KKorzuno** [comment]
Well it took me a long while and a lot of reading, but I got to the bottom of this. The classes for Containers and Software Systems have different annotations on them. To be exact: Software Systems have the @Nonnull annotations on addContainer methods, while Containers do not have addComponent methods. Is there any reason for this?

**Comment by simonbrowndotje** [comment]
It was a part of PR #98.

> Is there any reason for this?

Probably just an oversight. Feel free to address with a PR.

**Comment by KKorzuno** [comment]
A reason for a difference, not why it is there :)

**Comment by simonbrowndotje** [comment]
As I said in the previous comment, probably just an oversight ... you'd need to ask @klu2.

**Comment by klu2** [comment]
yes that's just an oversight.
